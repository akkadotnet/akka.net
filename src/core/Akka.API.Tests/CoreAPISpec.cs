//-----------------------------------------------------------------------
// <copyright file="CoreAPISpec.cs" company="Akka.NET Project">
//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
// </copyright>
//-----------------------------------------------------------------------

using System;
using System.IO;
using System.Linq;
using System.Runtime.CompilerServices;
using Akka.Cluster;
using Akka.Cluster.Tools.Singleton;
using Akka.Persistence;
using Akka.Remote;
using Akka.Streams.Dsl;
using ApiApprover;
using ApprovalTests;
using Mono.Cecil;
using Xunit;
using Akka.Persistence.Query;

namespace Akka.API.Tests
{
    public class CoreAPISpec
    {
        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveCore()
        {
            var assemblyPath = Path.GetFullPath(typeof(PatternMatch).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveRemote()
        {
            var assemblyPath = Path.GetFullPath(typeof(RemoteSettings).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApprovePersistence()
        {
            var assemblyPath = Path.GetFullPath(typeof(Persistent).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApprovePersistenceQuery()
        {
            var assemblyPath = Path.GetFullPath(typeof(PersistenceQuery).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveCluster()
        {
            var assemblyPath = Path.GetFullPath(typeof(ClusterSettings).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveClusterTools()
        {
            var assemblyPath = Path.GetFullPath(typeof(ClusterSingletonManager).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }
        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveStreams()
        {
            var assemblyPath = Path.GetFullPath(typeof(Source).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        static string Filter(string text)
        {
            return string.Join(Environment.NewLine, text.Split(new[]
            {
                Environment.NewLine
            }, StringSplitOptions.RemoveEmptyEntries)
                .Where(l => !l.StartsWith("[assembly: ReleaseDateAttribute("))
                .Where(l => !string.IsNullOrWhiteSpace(l))
                );
        }

    }
}
