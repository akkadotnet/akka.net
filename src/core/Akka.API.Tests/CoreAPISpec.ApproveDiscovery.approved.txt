[assembly: System.Reflection.AssemblyMetadataAttribute("RepositoryUrl", "https://github.com/akkadotnet/akka.net")]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace Akka.Discovery.Aggregate
{
    [Akka.Annotations.InternalApiAttribute()]
    public class AggregateServiceDiscovery : Akka.Discovery.ServiceDiscovery
    {
        public AggregateServiceDiscovery(Akka.Actor.ExtendedActorSystem system) { }
        public override System.Threading.Tasks.Task<Akka.Discovery.ServiceDiscovery.Resolved> Lookup(Akka.Discovery.Lookup query, System.TimeSpan resolveTimeout) { }
    }
    [Akka.Annotations.InternalApiAttribute()]
    public sealed class AggregateServiceDiscoverySettings
    {
        public AggregateServiceDiscoverySettings(Akka.Configuration.Config config) { }
        public System.Collections.Generic.List<string> DiscoveryMethods { get; }
    }
}
namespace Akka.Discovery.Config
{
    [Akka.Annotations.InternalApiAttribute()]
    public class ConfigServiceDiscovery : Akka.Discovery.ServiceDiscovery
    {
        public ConfigServiceDiscovery(Akka.Actor.ExtendedActorSystem system) { }
        public override System.Threading.Tasks.Task<Akka.Discovery.ServiceDiscovery.Resolved> Lookup(Akka.Discovery.Lookup lookup, System.TimeSpan resolveTimeout) { }
    }
    [Akka.Annotations.InternalApiAttribute()]
    public class ConfigServicesParser
    {
        public ConfigServicesParser() { }
        public static System.Collections.Generic.Dictionary<string, Akka.Discovery.ServiceDiscovery.Resolved> Parse(Akka.Configuration.Config config) { }
    }
}
namespace Akka.Discovery
{
    public class Discovery : Akka.Actor.IExtension
    {
        public Discovery(Akka.Actor.ExtendedActorSystem system) { }
        public Akka.Discovery.ServiceDiscovery Default { get; }
        public static Akka.Discovery.Discovery Get(Akka.Actor.ActorSystem system) { }
        public Akka.Discovery.ServiceDiscovery LoadServiceDiscovery(string method) { }
    }
    public class DiscoveryProvider : Akka.Actor.ExtensionIdProvider<Akka.Discovery.Discovery>
    {
        public DiscoveryProvider() { }
        public override Akka.Discovery.Discovery CreateExtension(Akka.Actor.ExtendedActorSystem system) { }
        public static Akka.Configuration.Config DefaultConfiguration() { }
    }
    public class Lookup : Akka.Actor.INoSerializationVerificationNeeded, System.IEquatable<Akka.Discovery.Lookup>
    {
        public Lookup(string serviceName, string portName = null, string protocol = null) { }
        public string PortName { get; }
        public string Protocol { get; }
        public string ServiceName { get; }
        public Akka.Discovery.Lookup Copy(string serviceName = null, string portName = null, string protocol = null) { }
        public bool Equals(Akka.Discovery.Lookup other) { }
        public override bool Equals(object obj) { }
        public override int GetHashCode() { }
        public static bool IsValid(string srv) { }
        public static Akka.Discovery.Lookup ParseSrv(string srv) { }
        public override string ToString() { }
        public Akka.Discovery.Lookup WithPortName(string portName) { }
        public Akka.Discovery.Lookup WithProtocol(string protocol) { }
    }
    public abstract class ServiceDiscovery
    {
        protected ServiceDiscovery() { }
        public abstract System.Threading.Tasks.Task<Akka.Discovery.ServiceDiscovery.Resolved> Lookup(Akka.Discovery.Lookup lookup, System.TimeSpan resolveTimeout);
        public System.Threading.Tasks.Task<Akka.Discovery.ServiceDiscovery.Resolved> Lookup(string serviceName, System.TimeSpan resolveTimeout) { }
        public class Resolved : Akka.Actor.INoSerializationVerificationNeeded, Akka.Event.IDeadLetterSuppression, System.IEquatable<Akka.Discovery.ServiceDiscovery.Resolved>
        {
            public Resolved(string serviceName, System.Collections.Generic.IEnumerable<Akka.Discovery.ServiceDiscovery.ResolvedTarget> addresses) { }
            public System.Collections.Immutable.ImmutableList<Akka.Discovery.ServiceDiscovery.ResolvedTarget> Addresses { get; }
            public string ServiceName { get; }
            public bool Equals(Akka.Discovery.ServiceDiscovery.Resolved other) { }
            public override bool Equals(object obj) { }
            public override int GetHashCode() { }
            public override string ToString() { }
        }
        public class ResolvedTarget : Akka.Actor.INoSerializationVerificationNeeded, System.IEquatable<Akka.Discovery.ServiceDiscovery.ResolvedTarget>
        {
            public ResolvedTarget(string host, System.Nullable<int> port = null, System.Net.IPAddress address = null) { }
            public System.Net.IPAddress Address { get; }
            public string Host { get; }
            public System.Nullable<int> Port { get; }
            public bool Equals(Akka.Discovery.ServiceDiscovery.ResolvedTarget other) { }
            public override bool Equals(object obj) { }
            public override int GetHashCode() { }
            public override string ToString() { }
        }
    }
}